tmf/org.eclipse.tracecompass.tmf.core/src/org/eclipse/tracecompass/tmf/core/statistics/TmfStateStatistics.java

```diff
@@ -119,11 +119,14 @@ public void dispose() {
         }
         List<Long> times = new ArrayList<>();
         for (int i = 0; i < timeRequested.length; i++) {
-            if (timeRequested[i] > fTotalsStats.getCurrentEndTime()) {
+            if (timeRequested[i] < fTotalsStats.getStartTime()) {
+                times.add(fTotalsStats.getStartTime());
```

Comment:
It should ensure that the start time is added only once. For end time it's only once because it then immediately breaks out of the loop.

Reply:
The issue is that the result should have as many times between the given array and the resulting list. I tried to fill the resulting list with 0 and do a query2d only for one start and the rest of the times but it was not working correctly. Although, I am not sure why.

Reply:
@PatrickTasse @arfio can we resolve this?